mobile/lib/components/ra_navigation_shell.dart

```diff
@@ -0,0 +1,122 @@
+    RaRoutes.broadcasts: 'Audycje',
+    RaRoutes.about: 'O nas',
+  };
+
+  int _determineSelected() {
+    final index = _navigationItems.keys.toList().indexOf(
+          _navigationItems.keys.firstWhere(
+            (key) => key == state.fullPath,
+            orElse: () => RaRoutes.home,
+          ),
+        );
+    return max(index, 0);
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final burgerMenuIconController = useAnimationController(
+      duration: const Duration(milliseconds: 450),
+      reverseDuration: const Duration(milliseconds: 250),
+    );
+
+    final selectedPageIndex = _determineSelected();
+
+    return BlocProvider(
+      create: (_) => AudioHandlerCubit(),
+      child: AnnotatedRegion(
+        value: SystemUiOverlayStyle(
+          systemNavigationBarColor: context.colors.backgroundDark,
+        ),
+        child: Scaffold(
+          appBar: RaAppBar(
+            height: 75,
+            icon: Icon(
+              Icons.menu,
+              color: context.colors.highlightGreen,
+              size: 32,
+              semanticLabel: 'RA AppBar menu button',
+            ),
+            bottomSize: 5,
+            mainColor: context.colors.backgroundDark,
+            accentColor: context.colors.highlightGreen,
+            iconButton: IconButton(
+              onPressed: () => _scaffoldKey.currentState!.isEndDrawerOpen
+                  ? _scaffoldKey.currentState!.closeEndDrawer()
+                  : _scaffoldKey.currentState!.openEndDrawer(),
+              icon: AnimatedIcon(
+                icon: AnimatedIcons.menu_close,
+                progress: burgerMenuIconController,
+                color: context.colors.highlightGreen,
+                size: 32,
+                semanticLabel: 'RA AppBar menu button',
+              ),
+            ),
+            text: 'Radio\nAktywne',
+            iconPath: 'assets/ra_logo/RA_logo.svg',
+            titlePadding: const EdgeInsets.only(left: 4, top: 8, bottom: 16),
+            imageHeight: 40,
+          ),
+          body: SafeArea(
+            child: Scaffold(
+              key: _scaffoldKey,
+              backgroundColor: Colors.transparent,
+              drawerScrimColor: context.colors.drawerBackgroundOverlay,
+              onEndDrawerChanged: (isOpened) => isOpened
+                  ? burgerMenuIconController.forward()
+                  : burgerMenuIconController.reverse(),
+              endDrawer: RaBurgerMenu(
+                onItemClicked: burgerMenuIconController.reverse,
+                selectedIndex: selectedPageIndex,
+                navigationItems: _navigationItems,
+              ),
+              body: child,
+            ),
+          ),
+          bottomNavigationBar:
+              RaBottomNavigationBar(selectedPageIndex: selectedPageIndex),
+          bottomSheet: const Padding(
+            padding: RaPageConstraints.outerWidgetPagePadding,
+            child: RadioPlayerWidget(),
+          ),
```

Comment:
The problem likely occurs because `burgerMenuIconController.reverse()` is not being called when the burger menu is closed by selecting an item from the bottom navigation bar.

To fix this, we can probably use the `onTap()` method but that is just an idea of the top of my head.

Currently, when the burger menu is open and the user selects another page from the bottom navigation bar, the menu closes correctly. However, the icon does not change back to the burger menu icon as it should.

<p float="left">
  <img src="https://github.com/RadioAktywne/ra-app/assets/101600853/316ac147-a7d0-43c5-9fc9-290963c80740" width="200" />
  <img src="https://github.com/RadioAktywne/ra-app/assets/101600853/a945e8ef-20bd-4a47-8cab-8347f8c4931f" width="200" />
</p>

Reply:
Great catch, fortunately the fix was easy

Reply:
Works great now